doc/shared/sunmatrix/SUNMatrix_KokkosDense.rst

```diff
@@ -0,0 +1,239 @@
+   SUNDIALS Copyright Start
+   Copyright (c) 2002-2022, Lawrence Livermore National Security
+   and Southern Methodist University.
+   All rights reserved.
+
+   See the top-level LICENSE and NOTICE files for details.
+
+   SPDX-License-Identifier: BSD-3-Clause
+   SUNDIALS Copyright End
+   ----------------------------------------------------------------
+
+.. _SUNMatrix.Kokkos:
+
+The SUNMATRIX_KOKKOSDENSE Module
+================================
+
+.. versionadded:: 6.4.0
+
+The SUNMATRIX_KOKKOSDENSE :c:type:`SUNMatrix` implementation provides a data
+structure for dense and dense batched (block-diagonal) matrices using Kokkos
+:cite:p:`edwards2014kokkos,trott2022kokkos` and KokkosKernels
+:cite:p:`trott2021kokkos` to support variety backends including serial, OpenMP,
+CUDA, HIP, and SYCL. Since Kokkos is a modern C++ library, the module is also
+written in modern C++ (it requires C++14) as a header only library. To utilize
+this ``SUNMatrix`` users will need to include
+``sunmatrix/sunmatrix_kokkosdense.hpp``. More instructions on building SUNDIALS
+with Kokkos and KokkosKernels enabled are given in
+:numref:`Installation.CMake.ExternalLibraries`. For instructions on building and
+using Kokkos and KokkosKernels, refer to the
+`Kokkos <https://kokkos.github.io/kokkos-core-wiki/index.html>`_
+and `KokkosKernels <https://github.com/kokkos/kokkos-kernels/wiki>`_.
+documentation.
+
+.. _SUNMatrix.Kokkos.Usage:
+
+Using SUNMATRIX_KOKKOSDENSE
+----------------------------
+
+The SUNMATRIX_KOKKOSDENSE module is defined by the ``DenseMatrix`` templated
+class in the ``sundials::kokkos`` namespace:
+
+.. code-block:: cpp
+
+   template<class ExecutionSpace = Kokkos::DefaultExecutionSpace,
+            class MemorySpace = class ExecSpace::memory_space>
+   class DenseMatrix : public sundials::impl::BaseMatrix,
+                       public sundials::ConvertibleTo<SUNMatrix>
+
+To use the SUNMATRIX_KOKKOSDENSE module, we begin by constructing an instance of
+the Kokkos dense matrix e.g.,
+
+.. code-block:: cpp
+
+   // Single matrix using the default execution space
+   sundials::kokkos::DenseMatrix<> A{rows, cols, sunctx};
+
+   // Batched (block-diagonal) matrix using the default execution space
+   sundials::kokkos::DenseMatrix<> Abatch{blocks, rows, cols, sunctx};
+
+   // Batched (block-diagonal) matrix using the Cuda execution space
+   sundials::kokkos::DenseMatrix<Kokkos::Cuda> Abatch{blocks, rows, cols, sunctx};
+
+   // Batched (block-diagonal) matrix using the Cuda execution space and
+   // a non-default execution space instance
+   sundials::kokkos::DenseMatrix<Kokkos::Cuda> Abatch{blocks, rows, cols,
+                                                      exec_space_instance,
+                                                      sunctx};
+
+Instances of the ``DenseMatrix`` class are implicitly or explicitly (using the
+:cpp:func:`~DenseMatrix::Convert` method) convertible to a :c:type:`SUNMatrix`
```

Comment:
```suggestion
:cpp:func:`DenseMatrix::Convert` method) convertible to a :c:type:`SUNMatrix`
```
I'm assuming that the `~` was a typo.  If not, then please disregard this comment.

Reply:
Prefixing with `~` means the link text will only be the last component of the target. So this will link to the documentation for`DenseMatrix::Convert` but will appear as `Convert()` in the text.